sycl/source/detail/non_cached_kernel_lock.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/locked.hpp>
+#include <CL/sycl/detail/pi.hpp>
+
+#include <map>
+#include <mutex>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+class NonCachedKernelLock {
+public:
+  ~NonCachedKernelLock();
+
+  Locked<RT::PiKernel> lockKernel(RT::PiKernel &K);
+
+private:
+  static bool Compare(RT::PiKernel Lhs, RT::PiKernel Rhs);
```

Comment:
This function is not used anywhere, I assume it was left over from prior to switching to `std::less`.